src/market.rs

```diff
@@ -226,8 +255,13 @@ async fn run_once(
     url: String,
     regions: Vec<String>,
     rates: &Vec<server::RegionalRates>,
-    gb_rates: &Vec<GBRateCard>
+    gb_rates: &Vec<GBRateCard>,
+    address_whitelist: &Arc<String>,
+    address_blacklist: &Arc<String>,
 ) {
+    let address_whitelist_clone: Arc<String> = Arc::clone(&address_whitelist);
```

Comment:
this does deep copies, we want only shallow copies

Reply:
Done